app/models/ems_refresh.rb

```diff
@@ -130,6 +130,11 @@ def self.queue_merge(targets, ems, create_task = false)
     MiqQueue.put_or_update(queue_options) do |msg, item|
       targets = msg.nil? ? targets : msg.data.concat(targets)
       targets = uniq_targets(targets)
+      obj = targets.select { |t| t.kind_of?(ApplicationRecord) }
+      if obj.present?
+        obj.each { |i| _log.warn("queue_merge called with class: #{i.class.name}, id: #{i.id}") }
+        raise ArgumentError, "cannot call queue_merge with an activerecord object"
+      end
```

Comment:
In hindsight, I wonder if we should _allow_ objects since the other methods also allow that interface.  Most of the other methods use `get_target_objects` to convert the various ways to pass in things _to_ objects. Feels like we need a `get_target_class_id_pairs` that takes objects and convert them to class/id pairs/

Either way I think it's fine.  I'm good with the code as is, but for consistency, we may want to allow it and just do the right thing.  @agrare Opinions?

Reply:
That's effectively what this did the first time.

Reply:
IMO if it isn't used I think we should remove it from the other methods as well (assuming they aren't called from elsewhere obviously) to keep it as simple as possible

Reply:
Maybe also make `queue_merge` a private_class_method since it really is an internal only method

Reply:
Oh I didn't realize it was internal interface...in that case, yeah it should be private, and then the "consistency" doesn't matter, and so yeah, it should be just pure class/id pairs and if not then ArgumentError.

Reply:
Yeah it is only called by `EmsRefresh.queue_refresh` https://github.com/search?q=org%3AManageIQ+queue_merge&type=code